### ISSUE TITLE
The publish to the build asset registry task should fail if there are duplicates in the merged manifest

### ISSUE THREAD

**Issue by riarenas** [dotnet/arcade#13349]
- [ ] This issue is blocking 
- [ ] This issue is causing unreasonable pain

The publishing infrastructure will fail when there are multiple entries for the same package + version combination in the merged manifest. This doesn't surface until repos have set up a default channel, and the scenario fails during the publishing stage. 

We should consider making this fail earlier during the Publish to Build Asset Registry task if we find that the Merged manifest will end up with duplicate entries

### Release Note Category
- [ ] Feature changes/additions 
- [ ] Bug fixes
- [ ] Internal Infrastructure Improvements

**Comment by riarenas** [comment]
This is related to https://github.com/dotnet/arcade/issues/5853 as well, because the current message is hard to self-diagnose, as seen in https://github.com/dotnet/arcade/issues/13347

**Comment by epananth** [comment]
Is this going to be a prod con thing?

**Comment by riarenas** [comment]
I don't see why? this is about the regular publishing infra.

**Comment by epananth** [comment]
I don't think it is. I may be wrong. Wondering if I could take this up today as SSA.. Atleast to evaluate what to do.

**Comment by riarenas** [comment]
I don't consider this meeting the bar.

**Comment by epananth** [comment]
We still don't know, who will own publishing.

**Comment by epananth** [comment]
I think Engineering Services will according to this thread https://teams.microsoft.com/l/message/19:83953de1ddd34bcaa9ebaba54cb8eba4@thread.skype/1685654947885?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=147df318-61de-4f04-8f7b-ecd328c256bb&parentMessageId=1685654947885&teamName=.NET%20Eng%20Services&channelName=General&createdTime=1685654947885

**Comment by epananth** [comment]
there is no epic as such for this, there is one for backporting v3 to arcade 3.x but this will not fit in that
